### ISSUE TITLE
Improve initial EpochNumber computation in TimeClock

### ISSUE THREAD

**Issue by Freyskeyd** [agglayer/agglayer#33]
## Description

When a `TimeClock` is created, `current_block` and `current_epoch_number` are initialized to `0`.
As the `TimeClock` is a future, when it starts, it computes the `current_block` based on the genesis datetime.
After that, it calculates and stores the `current_epoch_number` using a regular `store`.

An improvement could be to switch from `store` to `compare_exchange` to acknowledge that the `current` is `0` meaning that the `TimeClock` hasn't been modified yet. If it does, the `TimeClock` should shut down with proper error in logs.

This ticket is low priority and is somehow blocked by Agglayer/agglayer-rs#32

## Acceptance Criteria

- Have tests that detect and acknowledge the validity of the initial `current_epoch_number`
